src/core/proto/mem_desc.h

```diff
@@ -81,135 +73,7 @@ class mem_desc {
         NOT_IN_USE(len);
         return LKEY_ERROR;
     }
-};
-
-class mem_desc_compose : public mem_desc {
-public:
-    mem_desc_compose()
-        : m_array_size(0)
-    {
-        atomic_set(&m_ref, 0);
-    }
-
-    void get(void)
-    {
-        int ref = atomic_fetch_and_inc(&m_ref);
-
-        if (ref == 0) {
-            for (int i = 0; i < m_array_size; ++i) {
-                m_array[i]->get();
-            }
-        }
-    }
-
-    void put(void)
-    {
-        int ref = atomic_fetch_and_dec(&m_ref);
-
-        if (ref == 1) {
-            for (int i = 0; i < m_array_size; ++i) {
-                m_array[i]->put();
-            }
-            delete this;
-        }
-    }
-
-    uint32_t get_lkey(mem_buf_desc_t *desc, ib_ctx_handler *ib_ctx, const void *addr, size_t len)
-    {
-        if (likely(m_array_size != 0)) {
-            return m_array[0]->get_lkey(desc, ib_ctx, addr, len);
-        } else {
-            return LKEY_ERROR;
-        }
-    }
-
-    void add_child(mem_desc *child)
-    {
-        if (likely(m_array_size < DATA_SOURCE_NR_MAX)) {
-            m_array[m_array_size++] = child;
-        }
-    }
-
-private:
-    mem_desc *m_array[DATA_SOURCE_NR_MAX];
-    int m_array_size;
-    atomic_t m_ref;
-};
-
-class zcopy_hugepage : public mem_desc, lock_spin {
```

Comment:
don't get me wrong - I love deleting code - but does this mean we give up here on ZC api for dpcp in this branch?

Reply:
No. Its just old unused stuff.

Reply:
Discussed